### DISCUSSION TITLE
How to Use this Package with Multiple Database Connections?

### DISCUSSION THREAD

**Discussion by heyitsmi** [spatie/laravel-permission#2746] [General]
I plan to use this package with multiple database connections. Let's say I have two connections: master and default.

I want to use the users and roles tables from the "master" database connection and other tables such as permissions, model_has_roles, model_has_permissions and role_has_permissions from the default database connection.

I have created User and Role models like this:

User.php
```php
<?php

namespace App\Models;

use App\Traits\HasUuid;
use Laravel\Sanctum\HasApiTokens;
use Spatie\Permission\Models\Role;
use Laravel\Jetstream\HasProfilePhoto;
use Spatie\Permission\Traits\HasRoles;
use Illuminate\Notifications\Notifiable;
use Laravel\Fortify\TwoFactorAuthenticatable;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Foundation\Auth\User as Authenticatable;

class User extends Authenticatable
{
    use HasApiTokens;
    use HasFactory;
    use HasProfilePhoto;
    use Notifiable;
    use TwoFactorAuthenticatable;
    use HasRoles;
    use HasUuid;

    protected $connection = 'master';
}
```

Role.php
```php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Spatie\Permission\Models\Permission;
use Spatie\Permission\Models\Role as ModelsRole;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Role extends ModelsRole
{
    use HasFactory;

    protected $connection = 'master';

    protected $fillable = [
        'name',
        'guard_name',
    ];
}
```

![image](https://github.com/user-attachments/assets/43a63645-26f5-446d-95bf-6ebd2f5c2689)

With the method above, I have successfully retrieved the users and roles data from the "master" database connection, but the permissions loaded are also from the "master" database connection.

![image](https://github.com/user-attachments/assets/70edb436-7468-4084-aeee-fd123201a015)

I want the permissions data retrieved from the "default" database connection.

How can I load the permissions data from the "default" database connection?

**Comment by erikn69** [comment]
https://github.com/spatie/laravel-permission/pull/2095#issuecomment-1127709836

**Reply by heyitsmi** [reply]
Thank you for taking the time to answer.

I have tried it
```php
'table_names' => [
    'roles' => env('MASTER_DB_DATABASE').'.roles',
    'permissions' => env('DB_DATABASE').'.permissions',
    'model_has_permissions' => env('DB_DATABASE').'.model_has_permissions',
    'model_has_roles' => env('DB_DATABASE').'.model_has_roles',
    'role_has_permissions' => env('DB_DATABASE').'.role_has_permissions',
],
```

but I get an error like this
![image](https://github.com/user-attachments/assets/5affa9b3-7dd2-4f03-9763-55676b76a1ae)

**Reply by erikn69** [reply]
is It MariaDB or MySql?

**Reply by heyitsmi** [reply]
pgsql

**Reply by erikn69** [reply]
>pgsql

You didn't specify that, I have no idea if it can be done there,
I have tried it in MySQL and MariaDB and it works without problems

**Comment by josespinal** [comment]
Hi @hilmihidyt ,

Did you solve this?
